sunbeam-python/sunbeam/commands/proxy.py

```diff
@@ -0,0 +1,271 @@
+# Copyright (c) 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+
+import click
+import yaml
+from rich.console import Console
+from rich.table import Column, Table
+
+from sunbeam.clusterd.service import (
+    ClusterServiceUnavailableException,
+    ConfigItemNotFoundException,
+)
+from sunbeam.commands.juju import UpdateJujuModelConfigStep
+from sunbeam.commands.openstack import UpdateOpenStackModelConfigStep
+from sunbeam.commands.sunbeam_machine import DeploySunbeamMachineApplicationStep
+from sunbeam.commands.terraform import TerraformInitStep
+from sunbeam.jobs.checks import DaemonGroupCheck, VerifyBootstrappedCheck
+from sunbeam.jobs.common import (
+    FORMAT_TABLE,
+    FORMAT_YAML,
+    BaseStep,
+    Result,
+    ResultType,
+    Status,
+    convert_proxy_to_model_configs,
+    get_proxy_settings,
+    run_plan,
+    run_preflight_checks,
+    update_config,
+)
+from sunbeam.jobs.deployment import PROXY_CONFIG_KEY, Deployment
+from sunbeam.jobs.juju import CONTROLLER_MODEL, JujuHelper
+from sunbeam.jobs.manifest import Manifest
+from sunbeam.jobs.plugin import PluginManager
+from sunbeam.jobs.questions import (
+    ConfirmQuestion,
+    PromptQuestion,
+    QuestionBank,
+    load_answers,
+    write_answers,
+)
+
+LOG = logging.getLogger(__name__)
+console = Console()
+
+
+def _update_proxy(proxy: dict, deployment: Deployment):
+    client = deployment.get_client()
+
+    preflight_checks = [DaemonGroupCheck(), VerifyBootstrappedCheck(client)]
```

Comment:
This will not work in maas deployments (failure about ownership). I think we might need to have specilized `<<cluster is usable>>` checks. (Not in this PR)

Reply:
ack, will do in separate PR.